mevolib/fetch/BioSeqs.py

```diff
@@ -273,84 +217,77 @@ def from_seqfile (cls, seqfile, fileformat):
         seq_dict = {}
-        # Execute Entrez.esearch() to get the total number of sequences that
-        # matches the query in the Entrez database
+        # Execute Entrez.esearch() to get the total number of sequences that matches the query in the Entrez 
+        # database
         handle = Entrez.esearch(db=entrez_db, term=query, rettype='count')
         available_seqs = int(Entrez.read(handle)['Count'])
         handle.close()
         num_seqs = min(max_fetch, available_seqs)
-        if (available_seqs < 1):
+        if available_seqs < 1:
             warnings.warn('The query provided didn\'t return any sequence')
-        elif (max_fetch < 1):
+        elif max_fetch < 1:
             warnings.warn('"max_fetch" forces to create an empty dictionary')
         else: # num_seqs >= 1
             # Execute again Entrez.esearch() giving the total number of
             # sequences to get the complete list of Entrez database's sequence
             # identifiers
             sequence_ids = []
             for index in range(0, num_seqs, MAX_NUM_SEQS):
-                handle = Entrez.esearch(db=entrez_db, term=query, restart=index,
-                                        retmax=num_seqs)
+                handle = Entrez.esearch(db=entrez_db, term=query, restart=index, retmax=num_seqs)
                 record = Entrez.read(handle)
                 handle.close()
                 sequence_ids += record['IdList']
             # Fetch the first sequence and estimate the batch size
-            fetch_handle = Entrez.efetch(db=entrez_db, id=sequence_ids[0],
-                                         retmode='text', rettype=db_rettype)
+            fetch_handle = Entrez.efetch(db=entrez_db, id=sequence_ids[0], retmode='text', rettype=db_rettype)
             record_str = fetch_handle.read()
             fetch_handle.close()
-            record = SeqIO.read(StringIO(record_str), 'genbank')
+            record = SeqIO.read((record_str), 'genbank')
             seq_dict[record.id] = record
             batch_size = _estimate_batch_size(record_str)
-            # In batches of 'batch_size', fetch the Entrez database information
-            # of each sequence in text format through Entrez.efetch()
+            # In batches of 'batch_size', fetch the Entrez database information of each sequence in text 
+            # format through Entrez.efetch()
             start = 1
             exceptRaised = False
-            while (start < num_seqs):
+            while start < num_seqs:
                 end = start + batch_size
                 try:
                     fetch_handle = Entrez.efetch(
-                            db=entrez_db, id=sequence_ids[start:end],
-                            retmode='text', rettype=db_rettype)
+                                                 db=entrez_db, id=sequence_ids[start:end],
+                                                retmode='text', rettype=db_rettype)
```

Comment:
```suggestion
                    fetch_handle = Entrez.efetch(db=entrez_db, id=sequence_ids[start:end], retmode='text',
                                                 rettype=db_rettype)
```
And delete the previous line.